internal/provider/data_source_http.go

```diff
@@ -162,12 +163,29 @@ func (d *httpDataSource) Read(ctx context.Context, req datasource.ReadRequest, r
 
 	caCertificate := model.CaCertificate
 
-	tr := &http.Transport{
-		TLSClientConfig: &tls.Config{},
+	tr, ok := http.DefaultTransport.(*http.Transport)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Error configuring http transport",
+			"Error http: Can't configure http transport.",
+		)
+		return
+	}
+
+	// Prevent issues with tests modifying shared transport.
+	clonedTr := tr.Clone()
+
+	// Prevent issues with tests caching the proxy configuration.
+	clonedTr.Proxy = func(req *http.Request) (*url.URL, error) {
+		return httpproxy.FromEnvironment().ProxyFunc()(req.URL)
+	}
+
+	if clonedTr.TLSClientConfig == nil {
```

Comment:
Nit: To further prevent any differences due version to 3.2.0 changes, it is mentioned in the [`TLSClientConfig` documentation](https://pkg.go.dev/net/http#Transport.TLSClientConfig) that:

> // If nil, the default configuration is used.
> // If non-nil, HTTP/2 support may not be enabled by default.

I think we may want to consider moving this `nil` check inside each of the `!IsNull()` branches below to prevent any sort of further HTTP/2 oddities when there is no custom TLS configuration.